Python/chapter02/2.8 - Loop Detection/miguel_soln_2.8.py

```diff
@@ -171,20 +171,18 @@ def loop_detection(ll: LinkedList) -> Optional[Node]:
     A linked list is 'circular' when a node's
     next pointer points to an earlier node, so
     as to make a loop in the linked list.
-    Runtime:  O(n^2)
-    Space Complexity:  O(1)
+    Runtime:  O(n)
+    Space Complexity:  O(n)
```

Comment:
nice! this is much better _and_ simpler to understand.

stretch goal: there exists an O(1) space solution that does not rely on size.

Reply:
updated, O(1) space with O(n^2) runtime